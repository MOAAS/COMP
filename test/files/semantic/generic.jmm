class Simple {
    public int fun(int a, int b) {
        boolean c;
        Simple s;
        s = new Simple();

        if (a + b)
            c = true && false;
        else c = false;
        c = 5;
        a = 4 + c;

        while (s.ok(1,2)) {
            c = false;
        }

        return c;
    }

    public int ok(int a, int b) {
        return true;
    }

 }