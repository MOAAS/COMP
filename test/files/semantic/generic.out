Error: If expression is not a boolean. Found: int at line 7, column 12.
Error: Assignment of different types. Can't assign int to boolean at line 10, column 9.
Error: Right operand of + is not an int. Found: boolean at line 11, column 17.
Error: While expression is not a boolean. Found: int at line 13, column 15.
Error: Invalid return expression. Required: int, Provided: boolean at line 17, column 16.
Error: Invalid return expression. Required: int, Provided: boolean at line 21, column 16.
